modules/tinymce/src/plugins/table/test/ts/browser/command/MergeCellCommandTest.ts

```diff
@@ -35,68 +38,146 @@ UnitTest.asynctest('browser.tinymce.plugins.table.command.MergeCellCommandTest',
   const clearEvents = () => modifiedEvents = [];
   const defaultEvent = { type: 'tablemodified', structure: true, style: false } as TableModifiedEvent;
 
-  const testCommand = (editor: Editor, command: string, tests: MergeCellTest[]) => {
-    Tools.each(tests, (test) => {
-      clearEvents();
-      editor.getBody().innerHTML = test.before;
-      editor.selection.select(editor.dom.select('td[data-mce-selected]')[0], true);
-      editor.selection.collapse(true);
-      editor.execCommand(command);
-      LegacyUnit.equal(cleanTableHtml(editor.getContent()), test.after, test.message);
-      LegacyUnit.equal(modifiedEvents, test.expectedEvents);
-    });
+  const testMerge = (editor: Editor, test: MergeCellTest) => {
+    clearEvents();
+    editor.setContent(test.before, { format: 'raw' });
+    editor.selection.select(editor.dom.select('td[data-mce-selected]')[0], true);
+    editor.selection.collapse(true);
+    editor.execCommand('mceTableMergeCells');
+    assert.equal(cleanTableHtml(editor.getContent()), test.after);
+    assert.deepEqual(modifiedEvents, test.expectedEvents);
   };
 
   const cleanTableHtml = (html: string) => {
     return html.replace(/<p>(&nbsp;|<br[^>]+>)<\/p>$/, '');
   };
 
-  suite.test('TestCase-TBA: Table: mceTableMergeCells', (editor) => {
-    testCommand(editor, 'mceTableMergeCells', [
-      {
-        message: 'Should merge all cells into one',
-        before: (
-          '<table>' +
+  const getWidthNumber = (widthStyle: string): number => Number(widthStyle.replace('%', ''));
```

Comment:
Sugar dimension thing, or is this okay just being a search?

Reply:
The Sugar stuff is all computed (which with this particular case, shows a correct width), I need the raw CSS value to ensure that further operations will work i.e. inserting a column.

Reply:
I was particularly referencing this (potentially over engineered) module I wrote earlier: https://github.com/tinymce/tinymce/blob/develop/modules/sugar/src/main/ts/ephox/sugar/api/view/Dimension.ts

It just takes strings and returns numbers, basically. If you find it's hard to adopt, or overkill, as it's starting to feel in a lot of the places that I suggest it, I might think about redesigning or removing it in the future

Reply:
Oh cool I'll look at that